k8s/autopilot/autopilot.go

```diff
@@ -0,0 +1,33 @@
+package autopilot
+
+import (
+	autopilotclientset "github.com/libopenstorage/autopilot-api/pkg/client/clientset/versioned"
+	"k8s.io/client-go/rest"
+)
+
+// AutopilotRuleOps provides an interface to Autopilot operations.
+type AutopilotOps interface {
+	AutopilotRuleOps
+}
+
+func New(c autopilotclientset.Interface) *Client {
```

Comment:
+1

Reply:
That's not recommended to return an interface: https://github.com/golang/go/wiki/CodeReviewComments#interfaces
What the reason of doing this?

Reply:
This is interesting and good to know. Thanks for sharing. We have always been returning interfaces in New..() methods.
I am still wondering what good does it do to expose Client, especially here in this package where we have the interface as well as it's implementation. External consumers will mostly be assigning it to a Ops instead of Client instance.